src/api/app/controllers/webui/cloud/azure/upload_jobs_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Webui
+  module Cloud
+    module Azure
+      class UploadJobsController < ::Webui::Cloud::UploadJobsController
+        def new
+          xml_object = OpenStruct.new(params.slice(:project, :package, :repository, :arch, :filename))
+          @upload_job = ::Cloud::Backend::UploadJob.new(xml_object: xml_object)
+          @crumb_list.push << 'Azure'
+        end
+
+        def create
+          upload_params = permitted_params.merge(user: User.current, application_id: User.current.azure_configuration.application_id,
```

Comment:
You don't need to pass in the application_id and the application key. It is already in the upload_parameters (actually it is the only thing it contains :smile:) function which you call in the BackendApi. If you remove that, it is the same action as in Ec2 and you can leave it in the place where it was before :+1:

Reply:
I don't see from where is supposed to come... I'm assigning those here only not in somewhere else.

Reply:
Sure you do :smile: 
Here: https://github.com/openSUSE/open-build-service/pull/4835/files#diff-cd1d47a3124eb8adfbb326638f938d5cR17
You just get the azure_configuration from the user, the user is already passed in. The azure_configuration contains already the app_id and app_key :+1:

Reply:
Now I see it